### PR TITLE
Update ESLint and related devDependencies

### PR DESCRIPTION
**Pull request by kibertoad**
## What did you implement:

Updated ESLint and related devDependencies, addressed some minor linting issues.

## How can we verify it:

Run "lint" script.

## Todos:

- [X] Fix linting errors
- [X] Enable "Allow edits from maintainers" for this PR
- [X] Update the messages below

***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -8,7 +8,61 @@ module.exports = {
...
-    "import/no-extraneous-dependencies" : "off"
+    "import/no-extraneous-dependencies" : "off",
+    "prefer-destructuring": "off",
+    "comma-dangle": "off",
+    "no-buffer-constructor": "off",
+    "no-var": "off",
+    "prefer-template": "off",
+    "prefer-spread": "off",
+
+    "arrow-parens": "off",
```
Comment (kibertoad) [inline .eslintrc.js]: Probably at some point these new rules should be reviewed one by one and potentially fixed/restored, but it's going to be much easier after this PR.

**Comment by kibertoad** [comment]
Some interesting failures here. Will investigate further, probably some transitive dependencies affect this?..

**Comment by pmuens** [comment]
Hey @kibertoad thanks for taking the time to work on this PR and improve the codebase.

Unfortunately we have some inner-dependencies on older npm packages (e.g. we need a specific `sinon` version for `mocha`). That's why some of the tests are failing.

It's really great to see such improvements. Unfortunately we have to postpone such updates to the future. I'll close the PR and we'll re-visit it once we're in a spot where we can migrate the codebase to use newer dev dependencies.

Thanks again for taking the time to work on this! 👍 💯

**Comment by kibertoad** [comment]
@pmuens So pull request that would update these inner dependencies (such as `sinon`) wouldn't be accepted either?

**Pull request closed without merging**